### PR TITLE
Fix toolbar item cms

### PR DESCRIPTION
**Pull request by Ben0189**
#4795 

- Fix toolbar floating icon bug in rich text editor using container query in tailwind
- Refactor the width calculation to be maintainable approach
- Remove all magic number

### CODE REVIEW

**Review by Jord-Gui — APPROVED** [review]

**Review by Jord-Gui — CHANGES_REQUESTED** [review]
Fix resizing

**Comment by Ben0189** [comment]
Done fixing the resizing calculation, we will now safe from any width bug in the toolbar

**Review by Jord-Gui — CHANGES_REQUESTED** [review]
There's a small bug where the three dots pop up even though there isn't anything to hide
image

**Comment by Ben0189** [comment]
✅ Done fix - previous calculation does not take in consideration when embed button is not show 🥲 (now solved)

File: packages/tinacms/src/toolkit/fields/plugins/mdx-field-plugin/plate/components/fixed-toolbar-buttons.tsx [hunk 1]
```diff
@@ -100,14 +101,19 @@ const toolbarItems: { [key in ToolbarOverrideType]: ToolbarItem } = {
...
-  const [itemsShown, setItemsShown] = React.useState(11)
-  const { overrides } = useToolbarContext()
+  const [itemsShown, setItemsShown] = React.useState(10)
+  const { overrides, templates } = useToolbarContext()
+  const showEmbedButton = templates.length > 0
 
   useResize(toolbarRef, (entry) => {
     const width = entry.target.getBoundingClientRect().width
-    const itemsShown = (width - EMBED_ICON_WIDTH) / ICON_WIDTH
+    const paragraphIconWidth = width < MD_BREAKPOINT ? 58 : 128
```
Comment (JackDevAU) [inline packages/tinacms/src/toolkit/fields/plugins/mdx-field-plugin/plate/components/fixed-toolbar-buttons.tsx:110]: should we add these floating numbers into variables?

**Comment by Ben0189** [comment]
I had also refactor the width calculation function so it is now easier to add plugin without breaking the width again

**Review by Jord-Gui — APPROVED** [review]

**Pull request merged**